cmd/share.go

```diff
@@ -40,7 +40,18 @@ var shareCmd = &cobra.Command{
 		refType := fileref.FILE
 		statsMap, err := allocationObj.GetFileStats(remotepath)
 		if err != nil {
-			PrintError("Error in getting information about the object." + err.Error())
+			PrintError("Error fetching the file information", err)
+			os.Exit(1)
+		}
+
+		fileMeta, err := allocationObj.GetFileMeta(remotepath)
+		if err != nil {
+			PrintError("Error in getting the file meta of the object." + err.Error())
+			os.Exit(1)
+		}
+
+		if len(fileMeta.EncryptedKey) > 0 && (fflags.Changed("clientid") == false || fflags.Changed("encryptionpublickey") == false) {
+			PrintError("Clientid and/or encryptionpublickey are missing for the encrypted share!")
```

Comment:
This is not ideal, please use persistent flags like how it is done in other commands. Look at other commands' code to see how persistent flags work.